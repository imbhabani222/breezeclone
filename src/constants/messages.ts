const messages = {
  NOT_AUTHORIZED:
    "You are not authorized or session has been expired. Please try again.",
  NO_DATA_AVAILABLE: "No data available.",
  ACCESS_DENIED: "Access Denied/Forbidden.",
  DEFAULT_ERROR_MESSAGE:
    "There seems to be a problem, please retry after sometime.",
  USER_NOT_FOUND: "Invalid user",
  IMAGE_NOT_AVAILABLE: "Image not available",
  ARE_YOU_SURE: "Are you sure?",
  TYPE_OUTLINE: "outline",
  TYPE_RECTFILLED: "rectFilled",
  TYPE_FILLED: "filled",
  PLEASE_TYPE_ANSWER: "Please type your answer.",
  EXAM_NAME: "[0-2 years] Data Analytics",
  EXAM_TYPE: "By DoSelect UI/UX Testing",
  MODAL_CONTENT:
    "# Are you sure ? \n You still have **%hours% hours %minutes% mins %seconds% seconds** left to submit this assessment. </br></br>You have attempted %completed% of %total% questions.</br></br>You will not be able to attempt any question or make changes to your solution once you submit.",
  LOADING: "Loading...",
  PROCEED_TO_FEEDBACK: "PROCEED TO FEEDBACK PAGE",
  GOING_GREAT:
    "Going great .</br></br>You have attempted %completed% of %total% questions.",
  AUTO_NEXT_MODAL:
    "This section has been auto-submitted as the allotted time is up.</br></br>You have attempted %completed% of %total% questions.",
  BROWSER_TOLERANCE_INFO:
    "  You had navigated away from the test screen. This incident will be reported to the test administrator.",
  LOAD_MORE: "Load More",
  BACKBUTTON: " Go back to Assessment",
  PLATFORM_SUPPORT: "Platform Support",
  CLIENTS_SUPPORT: "Clients Support",
  HELP: "Help",
  QUESTION: "View All Questions",
  RIGHT: "right",
  FEEDBACK: {
    FEEDBACKCOMPLETED: "Woot! You've completed this assessment.",
    FEEDBACKTITLE: "SHARE FEEDBACK",
    FEEDBACKNOTIFIED:
      "We've notified team %orgName% about this. If you have any queries regarding this, please get in touch with your point of contact.",
    SENDFEEDBACK: "Send Feedback",
    ENTERFEWWORDS: "Please enter few words",
    MESSAGE_LABEL: "message",
    SUBMIT_LABEL: "Submit",
    FEEDBACKFORM: "feedbackForm",
    LIKETOTELL: "Is there something that you'd like to tell us?",
  },
  REPORT_LABLE: "REPORT AN ISSUE",
  REPORT_MODAL_TITLE: "Report an issue with Stack Using Interface",
  REPORT_MODAL_INPUT_LABEL: "  Other:",
  PROGRESS_DATABAR_TEXT: "%completed%/%total% Questions attempted",
  RUN_CODE_LABEL: "Run Code",
  ABORT_LABEL: "Abort",
  TEST_RESULTS_LABEL: "Test results",
  TEST_RESULTS_TEXT: 'Click on "Run code" to evaluate your solution',
  CUSTOM_INPUT_LABEL: "Custom Input",
  CUSTOM_INPUT_TEXT: "STDIN",
  ICON: "Icon",
  TIME_LABEL: {
    H: "H:",
    M: "M:",
    S: "S",
  },
  FAQ: {
    KEY1: "1",
    KEY2: "2",
    KEY3: "3",
    TAB1: "Help",
    TAB2: "Coding Environment",
    TAB3: "Status Code",
    SEARCH_PLACEHOLDER: "Ask a question",
    FAQINFOTEXT: `Please find below the general instructions and information about 
    DoSelect's assessment environment. We highly recommend that you go
    through this before you start taking the assessment.`,
  },
  TESTSTATUS: {
    ICONSTATUS: "iconsStatus",
    ACTIVATEICON: "activateIcon",
    DEACTIVATEICON: "deactivateIcon",
  },
  DESCRIPTION_LABEL: "Description",
  STATUS_CODE_LABEL: "Status Code",
  LIBRARIES_AVAILABLE_LABEL: "Libraries Available",
  TIME_LIMIT_LABEL: "Time Limit(in seconds)",
  MEMORY_UNIT_LABEL: "Memory Limit (in MB)",

  LIBRARIES_AVAILABLE_KEY: "Libraries available",
  TIME_LIMIT_KEY: "Time limit (in seconds)",
  MEMORY_UNIT_KEY: "Memory limit (in MB)",

  VERSION_LABEL: "Version",
  LANGUAGE_LABEL: "Language",
  DO_SELECT_RESEVE_RIGHTS: "Doselect. All rights reserved",
  LOGO_LABEL: "Logo",
  TEXT_LABEL: "text",
  TIME_FOR_EXAM_LABEL: "%hours% hours %minutes% mins",
  TIME_FOR_MODAL: "%hours% hours %minutes% mins %seconds% seconds",
  ALT_IMG: "icon Doselect",
  CANCEL: "Cancel",
  SUBMIT_END: "Submit and end the Assessment",
  SUBMIT_MOVE_NEXT: "Submit and Move to Next Section",
  PROCEED_TO_NEXT: "Proceed to next section",
  UPLOAD: "Upload",
  SAVE: "Save",
  QUESTION_TYPE: "MCQ + Coding + Fill in the  Blanks",
  LOGO: "do select logo",
  CAMERA: "Video Proctoring",
  NETWORK: "Network quality",
  TIMER: "Timer",

  BROWSER_FULLSCREEN: `<p>This %type% requires your browser window to be in fullscreen mode</p>`,
  UPLOAD_ATTACHMENT: "Upload an attachment with this solution.",
  VIDEO_TEST: {
    DESC: "Description",
    VID_DUR: "Video Duration",
    SEC: "seconds",
    RECORD: "Record",
    STOP: "Stop",
    RESUME: "Resume",
    PAUSE: "Pause",
    SUBMIT: "Submit",
    REC_DUR: "Record duration:",
  },

  CODING: {
    CODING: "Coding",
    RELOAD_STUB: "Reload Code Stub?",
    RELOAD_STUB_CONTENT: `You have changed the technology of this solution. Do you want
    to REPLACE your current code with the default code stub of
    this technology? Please note that you will lose your current
    code by doing this.`,
    NO_TEST_FOUND: "No Test Cases Found",
    RUN_ALL_TEST_CASES: "Run all Test Cases",
    DESCRIPTION: "Description",
    DESCRIPTIONS: "Descriptions",
    INPUT: "Input",
    OUTPUT: "Output",
    EXPECTED_OUTPUT: "Expected Output",
    EVAL_TIME: "Eval Time",
    CPU: "CPU",
    MEMORY: "Memory",
    EXECUTION_LOGS: "Execution Logs",
    ANNOTATION: "Annotation",
  },
};

export default messages;
