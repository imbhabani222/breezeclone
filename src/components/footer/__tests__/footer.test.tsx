import React from "react";
import { cleanup, render, screen } from "@testing-library/react";
import Footer from "../footer";
import messages from "../../../constants/messages";

const { DO_SELECT_RESEVE_RIGHTS } = messages;
const date: number = new Date().getFullYear();

describe("Footer", () => {
  function renderFooter(props: typeof date = date) {
    return render(<Footer />);
  }

  afterEach(cleanup);
  test("Render without Crashing", () => {
    renderFooter();
  });
  test("Match snapshot footer component", () => {
    const { asFragment } = renderFooter();
    expect(asFragment()).toMatchSnapshot();
  });
  // test("display current date", async () => {
  //   renderFooter();
  //   expect(screen.getByText(date)).toBeInTheDocument();
  // });
  test("display a Footer Content", async () => {
    renderFooter();
    expect(screen.getByTitle(DO_SELECT_RESEVE_RIGHTS)).toBeInTheDocument();
  });
});
