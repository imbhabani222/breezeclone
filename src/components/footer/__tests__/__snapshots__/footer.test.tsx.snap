// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Match snapshot footer component 1`] = `
<DocumentFragment>
  <footer
    class="footercl"
    title="Doselect. All rights reserved"
  >
    <div
      class="ant-typography footer_text"
    >
      Â© 2022 Doselect. All rights reserved
    </div>
  </footer>
</DocumentFragment>
`;
